### ISSUE TITLE
Hacked

### ISSUE THREAD

**Issue by VK-VZ** [UnamSanctam/UnamWebPanel#318]
![Screenshot_20240109_173044_Chrome](https://github.com/UnamSanctam/UnamWebPanel/assets/149912866/2150c591-ef28-4e07-bccd-a1550518c830)

For the 2nd time already.  The same hacker.  He steals workers.  This is the same hacker who hacked me 2 weeks ago.  The same wallet and api endpoint.  2 weeks ago I closed access to viewing php errors.  I also disabled error logging.  Now __UNAM_LIB/Logs errors are not recorded for me at all.  But they still hacked.  What to do? I saw a similar topic but didn’t understand what solution was found?

**Comment by UnamSanctam** [comment]
It is from a XSS attack (he doesn't actually have access to your web panel), you can find a version here that mitigates XSS: https://github.com/UnamSanctam/UnamWebPanel/issues/313#issuecomment-1882005200

**Comment by VK-VZ** [comment]
As I understand it, I need to remove the old panel and install a new one?  How can I get my old workers back?  Do I need to save the db folder and then return it to the new panel?

**Comment by UnamSanctam** [comment]
Yes just replace all the files (remove them first and then place the new ones there). If you have a fresh database then all your miners will appear again the next time they connect, or you can use your old database (`db/unamwebpanel.db`) by saving it and then overwriting your new web panel db file with your old one.

**Comment by VK-VZ** [comment]
I'm uploading a new panel to my server and now I can't login.  Incorrect password.  Although I changed the password 10 times and tried to log in.  What could be the reason?

**Comment by UnamSanctam** [comment]
Make sure the password set in the config.php is correct, and make sure you don't have caps lock or anything like that enabled when you're entering the password on the login page. You can also try copy-pasting the password from the config and see if it works. The web panel version has been tested and should work fine.

**Comment by VK-VZ** [comment]
I've done everything.  I completely removed the old panel and uploaded a new panel to the server.  I replaced only unamwebpanel.db from the old panel.  Is everything correct?  


![Screenshot_20240109_181913_My Files](https://github.com/UnamSanctam/UnamWebPanel/assets/149912866/fff94ffd-b0a7-46db-950c-cf5f3076cfed)

What are the 2 files?  from also transfer from the old panel to the new one?

**Comment by UnamSanctam** [comment]
Yes that should be correct, the two files are the write-ahead log and the index of the write-ahead log, they are used to make sure that the database does not get corrupted, you should copy all of those to your new web panel. Then when you log into the web panel remove the malicious miner entry that was added, it should be easily identifiable since it has nearly no data.

**Comment by VK-VZ** [comment]
"remove the malicious miner entry that was added" - I didn't understand what it was.  Can you show me a screenshot of what it is so I can delete it?  I just don't understand well

**Comment by UnamSanctam** [comment]
When you log into your web panel one of your miners there should almost have no information in it, that should be the "miner" entry with the attack in it. Or if it's a more sophisticated entry then it might have some information, in which case remove one that you don't recognize (maybe the latest one).

**Comment by VK-VZ** [comment]
@
![Screenshot_20240109_185551_Chrome](https://github.com/UnamSanctam/UnamWebPanel/assets/149912866/7feff84c-a7b6-4382-a854-cb94d4292d60)
Is this what you meant?  Is there only one such record or do we need to search for them all?

**Comment by UnamSanctam** [comment]
Yes that is correct, you can see the script injection there, there might only be one or they could have contacted your web panel with more fake miner connections. You can try searching for `script` and see if any others appear.

**Comment by VK-VZ** [comment]
I found 5 pieces and deleted them.  Thank you very much Unam.

**Comment by gabjohn3** [comment]
remember to use the last version here in #317 to secure ur webpanels

**Comment by Bendy127** [comment]
So sorry about what happened same thing happened to me my wallet got hacked and about 0.9 btc was stolen thanks to Mr John a whitehathacker who helped me recover all of it message him on telegram too t.me/Whitehathacker217
